app/main.py

```diff
@@ -1,8 +1,35 @@
 class Person:
-    # write your code here
-    pass
+
+    people = {}
+
+    def __init__(
+            self,
+            name: str,
+            age: int,
+    ):
+        self.name = name
+        self.age = age
+        Person.people[self.name] = self
 
 
 def create_person_list(people: list) -> list:
-    # write your code here
-    pass
+    result = []
+
+    for person in people:
+        Person(person["name"], person["age"])
+
+    for person in people:
+        our_object = Person.people[person["name"]]
+
+        if "wife" in person and person["wife"] is not None:
+            our_object.wife = Person.people[person["wife"]]
+            result.append(our_object)
+
+        elif "husband" in person and person["husband"] is not None:
```

Comment:
is not None dont nessesary write, because -  person["husband"] not in dict  , condition dotn executed code

Reply:
but it can be in dict as None(e.g. {"name": "Rachel", age: 30, "husband": None})